routes/books.js

```diff
@@ -52,12 +53,22 @@ router.route("/:id")
     })
   })
   .put(function (req, res) {
-    res.statusCode = 200;
-    res.end();
+    let book = req.body;
+    db.books.update(book).then((result) => {
+      if (result[0].affectedRows === 0) {    // If no rows are affected, send a 404.
+        res.statusCode = 404;
+        res.end();
+        return;
+      }
+       res.statusCode = 200;
+       res.end();
+     });
   })
-  .delete(function (req, res) {
-    res.statusCode = 200;
-    res.end();
+  .delete(function (req, res) {    
+    db.books.delete(req.params.id).then(function (book) {
```

Comment:
The function style here is inconsistent with the rest. Most of the time, you're using `(var) => {}`, but here you use `function(var) {}`. I'd stick to one or the other for these functions.